pkg/controllers/disruption/drift.go

```diff
@@ -82,16 +82,35 @@ func (d *Drift) ComputeCommand(ctx context.Context, candidates ...*Candidate) (C
 		consolidationTypeLabel: d.ConsolidationType(),
 	}).Set(float64(len(candidates)))
 
+	// Do a quick check through the candidates to see if they're empty.
+	// For each candidate that is empty with a nodePool allowing its disruption
+	// add it to the existing command.
+	empty := make([]*Candidate, 0, len(candidates))
```

Comment:
I see this code in drift and expiration. Is there a reason to do this per controller vs just moving emptiness disruption evaluation higher https://github.com/kubernetes-sigs/karpenter/blob/534b2711a96130aff85b0addb7ca0173f6135831/pkg/controllers/disruption/controller.go#L72?

I assume atm disruption controller runs through all the methods in order https://github.com/kubernetes-sigs/karpenter/blob/534b2711a96130aff85b0addb7ca0173f6135831/pkg/controllers/disruption/controller.go#L127

Reply:
+1. I think that we should think about how to coalesce a lot of this going forward. Feels out of scope of this particular PR but definitely something that we should do

Reply:
Unless I am misunderstanding that a node that is drifted/ expired + empty is at a higher priority than just an empty node.

Reply:
Agreed. Something we definitely want to do, which could be blended into the custom termination reason as well.